workshops/memory game/README.md

```diff
@@ -0,0 +1,376 @@
+
+Here, [getAttribute()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute) returns the value of a specified attribute on the element, [.push() method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) adds one or more elements to the end of an array and returns the new length of the array and [setTimeout()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) sets a timer which executes a function or specified piece of code once the timer expires.
+
+```javascript
+document.addEventListener('DOMContentLoaded', () => {
+  const cardArray = [....]
+  const board = document.querySelector('.board')
+  const placeholder = "https://cloud-5ystxzer7.vercel.app/7placeholder.png"
+  var cardsClicked = []
+  var cardsClickedId = []
+
+  //creating game board
+    function createBoard() {
+      for (let i = 0; i < cardArray.length; i++) {
+        var card = document.createElement('img')
+        card.setAttribute('src', placeholder)
+        card.setAttribute('data-id', i)
+        card.addEventListener('click', flipCard)
+        board.appendChild(card)
+      }
+    }
+
+  //flip  the card
+    function flipCard() {
+      var cardId = this.getAttribute('data-id')
+      cardsClicked.push(cardArray[cardId].name)
+      cardsClickedId.push(cardId)
+      this.setAttribute('src', cardArray[cardId].img)
+      if (cardsClicked.length ===2) {
+        setTimeout(checkForMatch, 500)
+      }
+    }
+  createBoard()
+})
+```
+
+Don't forget to uncomment the event-listener of the card.
+Comment the `if` statement in `flipCard` function and check weather the images are changing or not. The output works like this.
+
+![Flip card](https://cloud-odqx6kfb8.vercel.app/0flipcard.gif)
+
+Until now, we created a flippable game board.
+
+![Wooo!!](https://cloud-bos4syje4.vercel.app/0woo__.gif)
+
+Now that we have flipping cards, let’s handle the matching logic.
+
+## Match Card
+
+When we click the first card, it needs to wait until another card is flipped. So now, when the user clicks the second car,w e will check to see if it’s a match. In order to do that, let’s create a function `checkForMatch()`. Inside the function, select all the images we created using `querySelectorAll()` and define a `cards` array. Also get the two elements in `cardsClickedId` array into two variables `firstCard` and `secondCard` respectively. If the two cards match you get +1 to your score. These cards then disappear. If you click the same card again, a pop up alert notifies you and the cards flip back. If the next card you flip does not match, the cards flip back. The game continues until you match all the cards on the board.
+
+In place of the empty card, a blank image is displayed. Create a constant `blank` for the blank image.
+
+```javascript
+const blank = 'https://cloud-5ystxzer7.vercel.app/6blank.png'
+```
+
+![blank](https://cloud-5ystxzer7.vercel.app/6blank.png)
```

Comment:
I personally recommend removing this image since the effect is demonstrated by the gif below